### ISSUE TITLE
Serialization broken on 52

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#51317]
### Describe the bug

I can't even export a single card on v52. It will return:

```
2024-12-15 21:21:27,825 ERROR serialization.cmd :: Error during deserialization: Exception extracting Dashboard 1 {:model "Dashboard", :table :report_dashboard, :id 1, :entity_id "xBLdW9FsgRuB2HGhWiBa_", :cause "Error extracting Dashboard 1"}
  caused by: Error extracting Dashboard 1 {:model "Dashboard", :parent-id 1, :id 1}
  caused by: Error exporting nested :model/DashboardCard {:model :model/DashboardCard, :parent-id 1}
  caused by: Error extracting DashboardCard 29 {:model "DashboardCard", :id 29, :skip true, :metabase.models.serialization/type :target-not-found}
  caused by: FK target not found {:model :model/Card, :id 25, :skip true, :metabase.models.serialization/type :target-not-found}
```

### To Reproduce

1) start v52 with a postgres sample DB
2) make a dashboard with an x-ray
3) then run the export via the CLI (guess that via the api is the same)

you'll hit the issue

### Expected behavior

Serialization should work

### Logs

```
luis@pop-os:~/Documents/labs/serialization$ docker exec -it metabase_origin java -jar /app/metabase.jar export /target
Warning: environ value jdk-21.0.5+11 for key :java-version has been overwritten with 21.0.5
2024-12-15 21:21:20,690 INFO metabase.util :: Maximum memory available to JVM: 7.6 GB
2024-12-15 21:21:21,702 INFO util.encryption :: Saved credentials encryption is DISABLED for this Metabase instance. 🔓 
 For more information, see https://metabase.com/docs/latest/operations-guide/encrypting-database-details-at-rest.html
2024-12-15 21:21:22,337 ERROR public-settings.premium-features :: Error validating token: Assert failed: Metabase DB is not yet set up
((requiring-resolve (quote metabase.db/db-is-set-up?)))
WARNING: infinite? already refers to: #'kixi.stats.math/infinite? in namespace: kixi.stats.core, being replaced by: #'clojure.core/infinite?
2024-12-15 21:21:24,256 INFO driver.impl :: Registered abstract driver :sql  🚚
2024-12-15 21:21:24,274 INFO driver.impl :: Registered abstract driver :sql-jdbc (parents: [:sql]) 🚚
2024-12-15 21:21:24,277 INFO metabase.util :: Load driver :sql-jdbc took 8.1 ms
2024-12-15 21:21:24,278 INFO driver.impl :: Registered driver :h2 (parents: [:sql-jdbc]) 🚚
2024-12-15 21:21:24,398 INFO driver.impl :: Registered driver :mysql (parents: [:sql-jdbc]) 🚚
2024-12-15 21:21:24,421 INFO driver.impl :: Registered driver :postgres (parents: [:sql-jdbc]) 🚚
2024-12-15 21:21:24,506 INFO util.namespaces :: Loading namespace: metabase.notification.payload.impl.alert
2024-12-15 21:21:24,508 INFO util.namespaces :: Loading namespace: metabase.notification.payload.impl.dashboard-subscription
2024-12-15 21:21:24,510 INFO util.namespaces :: Loading namespace: metabase.notification.payload.impl.system-event
2024-12-15 21:21:24,517 INFO util.namespaces :: Loading namespace: metabase.channel.impl.email
2024-12-15 21:21:24,525 INFO util.namespaces :: Loading namespace: metabase.channel.impl.http
2024-12-15 21:21:24,530 INFO util.namespaces :: Loading namespace: metabase.channel.impl.slack
2024-12-15 21:21:25,734 INFO metabase.core :: 
Metabase v1.52.2.6 (0ec00d0) 

Copyright © 2024 Metabase, Inc. 

Metabase Enterprise Edition extensions are PRESENT.

Usage of Metabase Enterprise Edition features are subject to the Metabase Commercial License.See https://www.metabase.com/license/commercial/ for details.
2024-12-15 21:21:25,750 INFO serialization.cmd :: Exporting Metabase to /target
2024-12-15 21:21:25,751 INFO db.setup :: Verifying postgres Database Connection ...
2024-12-15 21:21:25,881 INFO db.setup :: Successfully verified PostgreSQL 16.2 application database connection. ✅
2024-12-15 21:21:25,881 INFO db.setup :: Checking if a database downgrade is required...
2024-12-15 21:21:25,926 INFO liquibase.database :: Set default schema name to public
2024-12-15 21:21:26,178 INFO liquibase :: Parsed changelog file 'liquibase.yaml'
2024-12-15 21:21:26,233 INFO db.setup :: Running Database Migrations...
2024-12-15 21:21:26,233 INFO db.setup :: Setting up Liquibase...
2024-12-15 21:21:26,238 INFO liquibase.database :: Set default schema name to public
2024-12-15 21:21:26,333 INFO liquibase :: Parsed changelog file 'liquibase.yaml'
2024-12-15 21:21:26,344 INFO db.liquibase :: Updating liquibase table to reflect consolidated changeset filenames
2024-12-15 21:21:26,361 INFO liquibase.lockservice :: Successfully acquired change log lock
2024-12-15 21:21:26,362 INFO db.liquibase :: No migration lock found.
2024-12-15 21:21:26,363 INFO db.liquibase :: Migration lock acquired.
2024-12-15 21:21:26,388 INFO liquibase.lockservice :: Successfully released change log lock
2024-12-15 21:21:26,389 INFO db.setup :: Liquibase is ready.
2024-12-15 21:21:26,389 INFO db.liquibase :: Checking if Database has unrun migrations...
2024-12-15 21:21:26,399 INFO liquibase.database :: Set default schema name to public
2024-12-15 21:21:26,457 INFO liquibase :: Parsed changelog file 'liquibase.yaml'
2024-12-15 21:21:26,478 INFO liquibase.changelog :: Reading from public.databasechangelog
2024-12-15 21:21:26,563 INFO db.liquibase :: No unrun migrations found.
2024-12-15 21:21:26,564 INFO db.setup :: Database Migrations Current ... ✅
2024-12-15 21:21:26,565 INFO metabase.util :: Database setup took 814.3 ms
2024-12-15 21:21:26,597 INFO public-settings.premium-features :: Checking with the MetaStore to see whether token 'f325...d01a' is valid...
2024-12-15 21:21:27,718 INFO models.serialization :: Extracting Dashboard 1 xBLdW9FsgRuB2HGhWiBa_
2024-12-15 21:21:27,825 ERROR serialization.cmd :: Error during deserialization: Exception extracting Dashboard 1 {:model "Dashboard", :table :report_dashboard, :id 1, :entity_id "xBLdW9FsgRuB2HGhWiBa_", :cause "Error extracting Dashboard 1"}
  caused by: Error extracting Dashboard 1 {:model "Dashboard", :parent-id 1, :id 1}
  caused by: Error exporting nested :model/DashboardCard {:model :model/DashboardCard, :parent-id 1}
  caused by: Error extracting DashboardCard 29 {:model "DashboardCard", :id 29, :skip true, :metabase.models.serialization/type :target-not-found}
  caused by: FK target not found {:model :model/Card, :id 25, :skip true, :metabase.models.serialization/type :target-not-found}
```

### Information about your Metabase installation

```JSON
v52.2.6
```

### Severity

P1

### Additional context

Probably we broke references?

**Comment by paoliniluis** [comment]
EDIT: doesn't happen on 51

**Comment by paoliniluis** [comment]
long story short: serialization works fine, but what doesn't work fine is a cascading deletion of cards which end up with a corrupted dashboard that serialization tries to grab

**Comment by seletz** [comment]
I get exactly the same issue it seems:

```
2025-04-12 10:50:44,472 INFO models.serialization :: Extracting Dashboard 1 xBLdW9FsgRuB2HGhWiBa_
2025-04-12 10:50:44,484 ERROR serialization.api :: Error during serialization: Error extracting Dashboard 1 {:model "Dashboard", :table :report_dashboard, :id 1, :entity_id "xBLdW9FsgRuB2HGhWiBa_", :cause "Error extracting Dashboard 1"}
  caused by: Error extracting Dashboard 1 {:model "Dashboard", :parent-id 1, :id 1}
  caused by: Error extracting nested :model/DashboardCard {:model :model/DashboardCard, :parent-id 1}
  caused by: Error extracting DashboardCard 29 {:model "DashboardCard", :id 29, :skip true, :metabase.models.serialization/type :target-not-found}
  caused by: FK target not found {:model :model/Card, :id 25, :skip true, :metabase.models.serialization/type :target-not-found}
```

@paoliniluis your comment suggests there's some corruption going on in the dashboard -- is there a way to fix this?  I'm running version `1.53.4`.  Would you like me to open a new issue?

**Comment by paoliniluis** [comment]
@seletz I forgot what was this issue (I think it's the example dashboard). Please create a new issue if possible
